GNUmakefile

```
@@ -3,27 +3,7 @@ MAKEFLAGS += --jobs=1
 -include config.unix
 
 #Lua version detection
-
-ifeq ($(origin LUA_VERSION),undefined)
-  # Get Lua version directly from the LUA interpreter 
-  ifneq ($(origin LUA),undefined)
-    LUA_VERSION := $(shell $(LUA) -e "print((_VERSION or ''):match('Lua (%d+%.%d+)') or '')")
-  else
-    
-	# system lua if LUA isn't defined
-    LUA_VERSION := $(shell lua -e "print((_VERSION or ''):match('Lua (%d+%.%d+)') or '')")
-  endif
-  
-  # warning ifcouldn't detect the version
-  ifeq ($(LUA_VERSION),)
-    $(warning Could not automatically detect Lua version. Please set LUA_VERSION manually.)
-    
-	LUA_VERSION := 5.1
-  endif
-endif
-
-
-
+LUA_VERSION ?= $(shell $(if $(LUA),$(LUA),lua) -e "print((_VERSION or ''):match('Lua (%d+%.%d+)') or '5.1')")
```

Better, but I think we can even make it a bit cleaner. I think this will accomplish the same thing, with the added advantage of keeping the LUA value around:

```suggestion
LUA ?= lua
LUA_VERSION ?= $(shell $(LUA) -e 'print((_VERSION or "5.1"):sub(-3))')
```